libs/src/litmus_libs/tls.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Shared module for handling TLS configuration in Charmed Litmus."""
+
+import logging
+from typing import Callable, Optional
+
+from ops import Container
+
+from litmus_libs.models import TLSConfig
+
+logger = logging.getLogger(__name__)
+
+
+class Tls:
```

Comment:
we used to have a tls helper class called `CertHandler` and it turns out this is the third reincarnation of that concept.
Its core responsibility is synchronizing aka reconciling the certificates received over relation data with a container's filesystem.

How about we give this a more descriptive name?

ideas:
- TlsConfigDataReconciler
- TlsReconciler

I think `self.tls_reconciler.reconcile()` sounds good!

If you go for this, update the module name accordingly.